src/tasks/processIssueComment/probot-processIssueComment.js

```diff
@@ -15,35 +15,55 @@ const {
 
 const getSafeRef = require('./utils/git/getSafeRef')
 
-async function processAddContributor({
+async function processAddContributors({
     context,
     commentReply,
     repository,
     optionsConfig,
-    who,
-    contributions,
+    contributors,
     branchName,
 }) {
-    if (contributions.length === 0) {
-        context.log.debug('No contributions')
-        return commentReply.reply(
-            `I couldn't determine any contributions to add, did you specify any contributions?
-            Please make sure to use [valid contribution names](https://allcontributors.org/docs/en/emoji-key).`,
-        )
+
+    const usersAdded = []
+    const usersMissed = []
+
+    async function addContributor(who, contributions) {
+        if (contributions.length === 0) {
+            context.log.debug(`No contributions for ${who}`)
+            usersMissed.push(who)
+            return
+        }
+
+        // TODO: wrap this blog in a try catch, if one user fails, don't fail the whole pull request
+        const { name, avatar_url, profile } = await getUserDetails({
+            github: context.github,
+            username: who,
+        })
+        
+        await optionsConfig.addContributor({
+            login: who,
+            contributions,
+            name,
+            avatar_url,
+            profile,
+        })
+
+        usersAdded.push(who)
     }
-    const { name, avatar_url, profile } = await getUserDetails({
-        github: context.github,
-        username: who,
-    })
 
-    await optionsConfig.addContributor({
-        login: who,
-        contributions,
-        name,
-        avatar_url,
-        profile,
+    // TODO: throttle max paralle requests
+    const addContributorPromises = contributors.map(function(contributor) {
+        return addContributor((contributor.who, contributor.contributions))
     })
 
+    await Promise.all(addContributorPromises)
+
+    if (usersAdded === 0) {
+        return commentReply.reply(
+        `I couldn't determine any contributions for ${usersAdded.join(', ')}.
+        Did you specify any contributions? Please make sure to use [valid contribution names](https://allcontributors.org/docs/en/emoji-key).`,
+    )
```

Comment:
Is this block supposed to finish here or are the following lines in that `if` block no longer useful?